sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1024,7 +1024,12 @@ class Analyzer(
                 DataSourceV2Relation.create(table, Some(catalog), Some(ident)))
           }
           val key = catalog.name +: ident.namespace :+ ident.name
-          Option(AnalysisContext.get.relationCache.getOrElseUpdate(key, loaded.orNull))
```

Comment:
Why not just:
```
Option(AnalysisContext.get.relationCache.getOrElseUpdate(key, loaded.orNull)).map { rel =>
  rel.transform { case multi: MultiInstanceRelation => multi.newInstance() }
} 
```

Reply:
sorry just saw your comment after running the merge script.

We only need to refresh the attributes if we get the relation from the cache. Otherwise, it's already a fresh relation.

Reply:
that is correct, but IMO it's a little cleaner. Creating new attributes should be super cheap, and not worth having a orElses and foreachs.